components/IterationPicker.tsx

```diff
@@ -41,35 +46,38 @@ const Iterations = ({ id, selectedIteration, addIteration, removeIteration }: It
     }
   }, [id]);
 
-  const formatDate = (date: string): string => format(parseISO(date), 'do MMM yy');
+  const formatDate = (date: string): string =>
+    format(parseISO(date), 'do MMM yy');
 
   return (
-    <ButtonDropdown>
-      <ButtonDropdown.Item
-        main={!Boolean(selectedIteration)} //todo: add the selected iteration to state.
-        onClick={(): void => {
-          removeIteration('iterations', null);
-        }}
-      >
-        No Iteration
-      </ButtonDropdown.Item>
+    <Select
+      disableMatchWidth
+      width='200px'
+      value={String(selectedIteration?.number || -1)}
+      onChange={(val) => {
+        const number = Number(val);
+        if (number === -1) {
+          removeIteration();
+        } else {
+          addIteration(iterations.find((it) => it.number === number));
+        }
+      }}
+    >
+      <Select.Option value='-1'>All Iterations</Select.Option>
       {iterations.map(
-        (iteration: Iteration): JSX.Element => {
+        (iteration: Iteration, index: Number): JSX.Element => {
```

Comment:
I've been using number as lowercase - do you know what the difference is with the captilized type?